### PR TITLE
HDDS-9756. Verbose log messages regarding pipeline creation failure

### PR DESCRIPTION
**Pull request by prashantpogde**
## What changes were proposed in this pull request?

Changing the log message to warn level instead of error.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9756

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineProvider.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ List<DatanodeDetails> pickNodesNotUsed(REPLICATION_CONFIG replicationConfig,
               "data in healthy node set. Nodes required: %d Found: %d",
           metadataSizeRequired, dataSizeRequired, nodesRequired,
           healthyDNsWithSpace.size());
-      LOG.error(msg);
+      LOG.warn(msg);
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineProvider.java:103]: `pickNodesNotUsed` seems to be used only for RATIS/ONE pipeline.  Similar message is logged as error for RATIS/THREE pipeline elsewhere:

https://github.com/apache/ozone/blob/74de0705c916bb877eae44310f43a0d7069cdef9/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java#L280-L288

Can you please change that one, too?  (Maybe even refactor to get rid of the duplication.)

**Review by adoroszlai — COMMENTED** [review]
Thanks @prashantpogde for the patch.

Please update PR title to be more specific about the change.

**Review by hemantk-12 — APPROVED** [review]
Changes look good to me.

**Comment by adoroszlai** [comment]
Thanks @prashantpogde for updating the patch and the PR title.

**Pull request merged**